yaksh/views.py

```diff
@@ -494,20 +499,29 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
             # validation
             if 'assignment' in request.FILES:
                 assignment_filename = request.FILES.getlist('assignment')
+                qp = QuestionPaper.objects.get(id=questionpaper_id)
             for fname in assignment_filename:
-                if AssignmentUpload.objects.filter(
-                    assignmentQuestion=current_question,
-                    assignmentFile__icontains=fname, user=user).exists():
-                    assign_file = AssignmentUpload.objects.get(
-                    assignmentQuestion=current_question,
-                    assignmentFile__icontains=fname, user=user)
+                assignment_files = AssignmentUpload.objects.filter(
+                            assignmentQuestion=current_question,
+                            assignmentFile__icontains=fname, user=user,
+                            question_paper=questionpaper_id)
+                if assignment_files.exists():
+                    assign_file = assignment_files.get(
+                            assignmentQuestion=current_question,
+                            assignmentFile__icontains=fname, user=user,
+                            question_paper=questionpaper_id)
                     os.remove(assign_file.assignmentFile.path)
                     assign_file.delete()
                 AssignmentUpload.objects.create(user=user,
-                    assignmentQuestion=current_question, assignmentFile=fname
+                    assignmentQuestion=current_question, assignmentFile=fname,
+                    question_paper=qp
                     )
             user_answer = 'ASSIGNMENT UPLOADED'
             if not current_question.grade_assignment_upload:
+                new_answer = Answer(question=current_question, answer=user_answer,
```

Comment:
Uploaded Assignment are evaluated, so there is a chance of file read each time during evaluation.
So, why not simply read file content and store it in database, which will be very useful! Otherwise, every time I have to download to view!